test_mem/test_mem.py

```diff
@@ -0,0 +1,26 @@
+import random
+from bit_vector import BitVector
+from mem.mem import gen_mem, Mode
+
+
+def test_cb_functional_model():
+    """
+    Small test to show the usage of the memory functional model
+    """
+    DATA_WIDTH = 4
+    DATA_DEPTH = 4
+    mem = gen_mem(DATA_DEPTH, DATA_WIDTH)
+    mem_inst = mem()
+
+    config_addr = BitVector(0, 32)
+    config_data = BitVector(0, 32)
+    mem_inst.config[config_addr] = config_data
+    assert mem_inst.config[config_addr] == config_data
+    mem_inst.mode = Mode.SRAM
```

Comment:
we should just use the `config[]` interface rather than directly program the mode.

Reply:
Done via the new interface that mimics the low-level interface to config via __call__